galley/pkg/kube/converter/converter.go

```diff
@@ -68,3 +70,54 @@ func legacyMixerResource(_ resource.Info, name string, u *unstructured.Unstructu
 		Contents: s,
 	}, nil
 }
+
+func authPolicyResource(destination resource.Info, name string, u *unstructured.Unstructured) (string, time.Time, proto.Message, error) {
+	p, err := toProto(destination, u.Object["spec"])
+	if err != nil {
+		return "", time.Time{}, nil, err
+	}
+
+	policy, ok := p.(*authn.Policy)
+	if !ok {
+		return "", time.Time{}, nil, fmt.Errorf("object is not of type %v", destination.TypeURL)
+	}
+
+	// The pilot authentication plugin's config handling allows the mtls
+	// peer method object value to be nil. See pilot/pkg/networking/plugin/authn/authentication.go#L68
+	//
+	// For example,
+	//
+	//     metadata:
+	//       name: d-ports-mtls-enabled
+	//     spec:
+	//       targets:
+	//       - name: d
+	//         ports:
+	//         - number: 80
+	//       peers:
+	//       - mtls:
+	//
+	// This translates to the following in-memory representation:
+	//
+	//     policy := &authn.Policy{
+	//       Peers: []*authn.PeerAuthenticationMethod{{
+	//         &authn.PeerAuthenticationMethod_Mtls{},
+	//       }},
+	//     }
+	//
+	// The PeerAuthenticationMethod_Mtls object with nil field is lost when
+	// the proto is re-encoded for transport via MCP. As a workaround, fill
+	// in the missing field value which is functionality equivalent.
+	//
+	// Alternatively, it might also be possible to customize the marshaling
+	// code to preserve the nil object.
+	for i, peer := range policy.Peers {
+		if mtls, ok := peer.Params.(*authn.PeerAuthenticationMethod_Mtls); ok && mtls.Mtls == nil {
+			policy.Peers[i] = &authn.PeerAuthenticationMethod{
+				&authn.PeerAuthenticationMethod_Mtls{&authn.MutualTls{}},
+			}
+		}
+	}
+
```

Comment:
cc @ozevren 

The test code below demonstrates the problem. Pilot currently avoids this problem by decoding from the CRD (via JSON/YAML). The partially nil one-of is never proto encoded.

```bash
$ cat auth.go
package main

import (
        "fmt"

        "github.com/gogo/protobuf/proto"
        authn "istio.io/api/authentication/v1alpha1"
)

func main() {
        policy := &authn.Policy{
                Peers: []*authn.PeerAuthenticationMethod{{
                        &authn.PeerAuthenticationMethod_Mtls{},
                }},
        }
        encoded, err := proto.Marshal(policy)
        if err != nil {
                panic(err)
        }

        var decoded authn.Policy
        if err := proto.Unmarshal(encoded, &decoded); err != nil {
                panic(err)
        }

        fmt.Printf("before %+v\n", policy.GetPeers())
        fmt.Printf("decoded %+v\n", decoded.GetPeers())
}

$ go run auth.go
before [mtls:/* nil */ ]
decoded []

```